cli/pkg/acm/configsync.go

```diff
@@ -0,0 +1,277 @@
+// Source: https://stackoverflow.com/questions/21151714/go-generate-an-ssh-public-key
+func InitSSH() error {
+	privateKeyPath := "id_rsa"
+	pubKeyPath := "id_rsa.pub"
+
+	// make ssh dir if not exists
+	err := os.MkdirAll("ssh", 0700)
+	if err != nil {
+		log.Warnf("Error making ssh dir: %v", err)
+	}
+
+	// ssh keygen to local dir
+	log.Info("Generating private key")
+	privateKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		return err
+	}
+
+	// generate and write private key as PEM
+	log.Info("Encoding private key")
+	privateKeyFile, err := os.Create(privateKeyPath)
+	defer privateKeyFile.Close()
+	if err != nil {
+		return err
+	}
+	privateKeyPEM := &pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(privateKey)}
+	if err := pem.Encode(privateKeyFile, privateKeyPEM); err != nil {
+		return err
+	}
+
+	// generate and write public key
+	log.Info("Generating public key")
+	pub, err := ssh.NewPublicKey(&privateKey.PublicKey)
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(pubKeyPath, ssh.MarshalAuthorizedKey(pub), 0655)
+	return nil
+}
+
+func PromptUser(conf *config.Config) error {
+	// read public key as string
+	bytes, err := ioutil.ReadFile("id_rsa.pub")
+	if err != nil {
+		return err
+	}
+	pubKey := string(bytes)
+
+	// Prompt user to register ssh public key to their Cloud Source Repositories
+	log.Info("💻 Copy the key below to the clipboard, then open this link to register: https://source.cloud.google.com/user/ssh_keys")
+	log.Info(pubKey)
+	log.Info("Once you've registered the key, press enter to continue...")
+	fmt.Scanln() // wait for Enter Key
+
+	// Prompt user to run make start proxy
+	if conf.PrivateEndpoint {
+		log.Info("⚠️ Your clusters have Private Endpoints. Please open another terminal tab and run the following command to proxy via your GCE Bastion Host.")
+		log.Infof("gcloud beta compute ssh gke-tk-bastion --tunnel-through-iap --project %s --zone %s-b -- -4 -L8888:127.0.0.1:8888", conf.ClustersProjectID, conf.ClustersConfig[0].Region)
+		log.Info("Once the proxy tunnel is running, press enter to continue...")
+		fmt.Scanln() // wait for Enter Key
+	}
+	return nil
+}
+
+func GenerateKubeConfig(conf *config.Config) (*api.Config, error) {
+	projectId := conf.ClustersProjectID
+	log.Infof("Clusters Project ID is %s", projectId)
+	ctx := context.Background()
+
+	svc, err := container.NewService(ctx)
+	if err != nil {
+		return api.NewConfig(), fmt.Errorf("container.NewService: %w", err)
+	}
+
+	// Basic config structure
+	ret := api.Config{
+		APIVersion: "v1",
+		Kind:       "Config",
+		Clusters:   map[string]*api.Cluster{},  // Clusters is a map of referencable names to cluster configs
+		AuthInfos:  map[string]*api.AuthInfo{}, // AuthInfos is a map of referencable names to user configs
+		Contexts:   map[string]*api.Context{},  // Contexts is a map of referencable names to context configs
+	}
+
+	// Ask Google for a list of all kube clusters in the given project.
```

Comment:
Note for myself. We need to call out in the docs that projects used with the toolkit should be empty.